src/main/java/com/google/api/codegen/metacode/InitCodeNode.java

```diff
@@ -271,13 +272,14 @@ private void resolveNamesAndTypes(
 
       // Validate initValueConfig, or generate random value
       if (initValueConfig.hasSimpleInitialValue()) {
-        validateValue(type, initValueConfig.getInitialValue());
+        validateValue(type, initValueConfig.getInitialValue().getValue());
       } else if (initValueConfig.isEmpty()
           && type.isPrimitive()
           && !type.isRepeated()
           && valueGenerator != null) {
         String newValue = valueGenerator.getAndStoreValue(type, identifier);
-        initValueConfig = InitValueConfig.createWithValue(newValue);
+        initValueConfig =
+            InitValueConfig.createWithValue(new InitValue(newValue, InitValueType.Literal));
```

Comment:
There are lots of uses of "new InitValue(...)". What do you think about static factory methods, such as InitValue.createLiteral(String value) and InitValue.createVariable(String variableName), or something similar?